### PR TITLE
Refactored Identity and other modules + ECID not primary in IdentityMap by default

### PR DESCRIPTION
**Pull request by addb**
- Modularize code
- Cleanup and refactor Identity code
- Don't send ECID as `primary = true`
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/main/core/identityModule.brs [hunk 1]
```diff
@@ -79,17 +76,17 @@ function _adb_IdentityModule(configurationModule as object) as object
...
-                return
-            end if
-
             _adb_logVerbose("_saveECID() - Saving ECID:(" + FormatJson(ecid) + ") to presistence.")
             localDataStoreService.writeValue(_adb_InternalConstants().LOCAL_DATA_STORE_KEYS.ECID, ecid)
         end function,
 
+        _deleteECID: function() as void
+            localDataStoreService = _adb_serviceProvider().localDataStoreService
+            _adb_logDebug("_saveECID() - Removing ECID from persistence.")
```
Comment (yangyansong-adbe) [inline code/main/core/identityModule.brs:85]: "_deleteECID() - Removing ..."

File: code/main/core/identityModule.brs [hunk 2]
```diff
@@ -101,6 +98,36 @@ function _adb_IdentityModule(configurationModule as object) as object
...
             url = _adb_buildEdgeRequestURL(configId, _adb_generate_UUID(), edgeDomain)
+            jsonBody = m._getECIDQueryPayload()
+            networkResponse = _adb_serviceProvider().networkService.syncPostRequest(url, jsonBody)
+
+            return m._getECIDFromQueryResponse(networkResponse)
+        end function,
+
+        _getECIDFromQueryResponse: function(networkResponse as dynamic) as dynamic
+            if not _adb_isNetworkResponse(networkResponse)
+                _adb_logError("processRequests() - Edge response is invalid.")
```
Comment (yangyansong-adbe) [inline code/main/core/identityModule.brs:109]: "_getECIDFromQueryResponse() - Edge ..."

File: code/main/core/identityModule.brs [hunk 3]
```diff
@@ -53,16 +51,15 @@ function _adb_IdentityModule(configurationModule as object) as object
...
+                _adb_logVerbose("updateECID() - Saving ECID:(" + FormatJson(ecid) + ") in cache and persistence.")
+                m._saveECID(m._ecid)
             end if
-            _adb_logVerbose("updateECID() - Saving ECID:(" + FormatJson(ecid) + ") in cache and persistence.")
-            m._ecid = ecid
-            m._saveECID(m._ecid)
+
         end function,
 
         _loadECID: function() as dynamic
```
Comment (yangyansong-adbe) [inline code/main/core/identityModule.brs:65]: Not related to your change, can we check if the loaded ECID is not empty before returning?

File: code/main/core/identityModule.brs [hunk 4]
```diff
@@ -101,6 +98,36 @@ function _adb_IdentityModule(configurationModule as object) as object
...
+        end function,
+
+        _getECIDFromQueryResponse: function(networkResponse as dynamic) as dynamic
+            if not _adb_isNetworkResponse(networkResponse)
+                _adb_logError("processRequests() - Edge response is invalid.")
+                return invalid
+            end if
+
+            if  not networkResponse.isSuccessful() or networkResponse.getResponseString() = invalid
+                _adb_logError("_queryECID() - Error occured while quering ECID from service side. The response code: (" + FormatJson(responseJson.code) + ")")
```
Comment (yangyansong-adbe) [inline code/main/core/identityModule.brs:114]: "_getECIDFromQueryResponse() .."

File: code/main/core/identityModule.brs [hunk 5]
```diff
@@ -101,6 +98,36 @@ function _adb_IdentityModule(configurationModule as object) as object
...
+                return invalid
+            end if
+
+            responseJson = ParseJson(networkResponse.getResponseString())
+            if responseJson <> invalid and responseJson.handle[0] <> invalid and responseJson.handle[0].payload[0] <> invalid
+                _adb_logVerbose("_queryECID() - Received response with payload: (" + FormatJson(responseJson) + ").")
+                remote_ecid = responseJson.handle[0].payload[0].id
+                return remote_ecid
+            else
+                _adb_logError("_queryECID() - Error extracting ECID, invalid response from server.")
```
Comment (yangyansong-adbe) [inline code/main/core/identityModule.brs:124]: "_getECIDFromQueryResponse() .."

File: code/main/core/identityModule.brs [hunk 6]
```diff
@@ -101,6 +98,36 @@ function _adb_IdentityModule(configurationModule as object) as object
...
+            end if
+
+            if  not networkResponse.isSuccessful() or networkResponse.getResponseString() = invalid
+                _adb_logError("_queryECID() - Error occured while quering ECID from service side. The response code: (" + FormatJson(responseJson.code) + ")")
+                return invalid
+            end if
+
+            responseJson = ParseJson(networkResponse.getResponseString())
+            if responseJson <> invalid and responseJson.handle[0] <> invalid and responseJson.handle[0].payload[0] <> invalid
+                _adb_logVerbose("_queryECID() - Received response with payload: (" + FormatJson(responseJson) + ").")
```
Comment (yangyansong-adbe) [inline code/main/core/identityModule.brs:120]: "_getECIDFromQueryResponse() .."

**Review by yangyansong-adbe — APPROVED** [review]
LGTM, only a couple of minor comments.

File: code/main/core/identityModule.brs (hunk 3 above)
Comment (addb) [reply code/main/core/identityModule.brs:65]: I have refactored it and that would be the responsibility of the caller to check and log.

**Pull request merged**